src/widget-sources/widget-tileset-worker-source.ts

```diff
@@ -0,0 +1,238 @@
+  override async getCategories({
+    abortController,
+    ...options
+  }: CategoryRequestOptions): Promise<CategoryResponse> {
+    return this._executeWorkerMethod(
+      this.props.tableName,
+      Method.GET_CATEGORIES,
+      [options],
+      abortController?.signal
+    );
+  }
+
+  override async getScatter({
+    abortController,
+    ...options
+  }: ScatterRequestOptions): Promise<ScatterResponse> {
+    return this._executeWorkerMethod(
+      this.props.tableName,
+      Method.GET_SCATTER,
+      [options],
+      abortController?.signal
+    );
+  }
+
+  override async getTable({
+    abortController,
+    ...options
+  }: TableRequestOptions): Promise<TableResponse> {
+    return this._executeWorkerMethod(
+      this.props.tableName,
+      Method.GET_TABLE,
+      [options],
+      abortController?.signal
+    );
+  }
+
+  override async getTimeSeries({
+    abortController,
+    ...options
+  }: TimeSeriesRequestOptions): Promise<TimeSeriesResponse> {
+    return this._executeWorkerMethod(
+      this.props.tableName,
+      Method.GET_TIME_SERIES,
+      [options],
+      abortController?.signal
+    );
+  }
+
+  override async getRange({
+    abortController,
+    ...options
+  }: RangeRequestOptions): Promise<RangeResponse> {
+    return this._executeWorkerMethod(
+      this.props.tableName,
+      Method.GET_RANGE,
+      [options],
+      abortController?.signal
+    );
+  }
+
+  /////////////////////////////////////////////////////////////////////////////
+  // WEB WORKER MANAGEMENT
+
+  // TODO: Singleton? Pool shared among datasets? One per dataset?
+  protected static WORKER: Worker;
+  protected static _nextRequestID = 1;
+
+  static init() {
+    WidgetTilesetWorkerSource.WORKER = new Worker(
+      new URL('@carto/api-client/worker', import.meta.url),
+      {
+        type: 'module',
+        name: 'cartowidgettileset',
+      }
+    );
+  }
+
+  _executeWorkerMethod<T>(
+    tableName: string,
+    method: Method,
+    params: unknown[],
+    signal?: AbortSignal
+  ): Promise<T> {
+    const worker = WidgetTilesetWorkerSource.WORKER;
+    const requestId = WidgetTilesetWorkerSource._nextRequestID++;
+
+    // TODO: ViewState may contain non-serializable data, which we do not need.
+    // Remove this sanitization after sc-469614 is fixed.
+    const options = params[0] as any;
+    if (options?.spatialIndexReferenceViewState) {
+      const {zoom, latitude, longitude} =
+        options.spatialIndexReferenceViewState;
+      options.spatialIndexReferenceViewState = {zoom, latitude, longitude};
+    }
+
+    worker.postMessage({
+      requestId,
+      tableName,
+      method,
+      params,
+    } as WorkerRequest);
+
+    return new Promise((resolve, reject) => {
+      function listener(e: MessageEvent) {
+        const response = e.data as WorkerResponse;
+        if (response.requestId !== requestId) return;
+
+        worker.removeEventListener('message', listener);
+
+        if (signal?.aborted) {
```

Comment:
to be robust, we should also install `signal.onabort(cleanup)` to abort and unsubscribe immediately when we abort ... and not to wait for response and only then abort (not sure how long it will take)

Reply:
Ok, I like that idea thanks! Added.